torch_frame/data/mapper.py

```diff
@@ -129,7 +130,19 @@ def forward(
         if ser.dtype != 'object':
             raise ValueError('Multi-categorical types expect string as input')
         values = []
+        original_index = ser.index
         ser = ser.apply(self._row_to_tensor)
```

Comment:
we are not returning a tensor here, so maybe better to rename to `self._split_by_sep` or something similar.